src/rgw/motr/synchronization/rgw_motr_gc_sync.h

```diff
@@ -0,0 +1,47 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=2 sw=2 expandtab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * SAL implementation for the CORTX Motr backend
+ *
+ * Copyright (C) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+#ifndef __MOTR_GC_SYNCHROZATION__
+#define __MOTR_GC_SYNCHROZATION__
+
+#include "../../rgw_sal_motr.h"
+#include "motr/synchronization/rgw_motr_sync.h"
+
+class MotrGCThreadSync : public MotrSync {
+private:
+  std::shared_ptr<MotrLockProvider> motr_gclock;
+
+public:
+  MotrGCThreadSync(std::shared_ptr<MotrLockProvider> _motr_gclock)
+    :motr_gclock(_motr_gclock) {
+  }
+  virtual int lock(const std::string &lock_name, MotrLockType lock_type, utime_t lock_duration, const std::string locker_id);
+  virtual int unlock(const std::string &lock_name, MotrLockType lock_type, const std::string locker_id);
+};
+
+class MotrKVLockProvider : public MotrLockProvider {
+private:
+  const DoutPrefixProvider* _dpp;
+  rgw::sal::MotrStore* _store;
+  std::string _lock_index;
+public:
+  void initialize(const DoutPrefixProvider* dpp, rgw::sal::MotrStore* _s, std::string& lock_index_name);
+  virtual int read_lock(const std::string &lock_name, Motr_lock_info_t* lock_info);
+  virtual int write_lock(const std::string &lock_name, Motr_lock_info_t* lock_info );
+  virtual int remove_lock(const std::string &lock_name, const std::string& locker_id);
+};
+
+#endif
```

Comment:
Please include an empty line at the end.

Reply:
done